doc/design_AsyncNDRegisterAcessor_and_NumericAddressedBackend.dox

```diff
@@ -0,0 +1,158 @@
+namespace ChimeraTK {
+/**
+\page design_AsyncNDRegisterAccessor_NumericAddressed Design: AsyncNDRegisterAccessor
+
+\section  design_AsyncNDRegisterAccessor The AsyncNDRegisterAccessor and the AsyncAccessorManager
+
+The AsyncNDRegisterAccessor is the NDRegisterAccessor implementation for push-type data. It is used
+if accessors with AccessMode::wait_for_new_data are requested. See \ref transferElement_B_8
+"Technical specification: TransferElement B.8" for the requirements it is based on.
+
+The AsyncNDRegisterAccessor is a generic implementation for all backends and contains the following components
+
+- an NDRegisterAccessor<UserType>::Buffer which serves as a send buffer and contains the data value
+  (2D vector of UserType), a VersionNumber and a DataValidity flag.
+- a lock free queue to transport the data of the send buffer
+- functions to activate/deactivate the accessor, and to push data and exceptions into the queue
+- a (synchronous) accessor which is used for write operations (optional)
+
+One AsyncAccessorManager is used per interrupt that can occur. The AsyncAccessorManager has three main functions:
+
+- It serves as a factory for AsyncNDRegisterAccessors
+- It contains a list of created asynchronous accessors and a subscribe/unsubscribe mechanism
+- It provides functions to act on all asynchronous accessors
+
+\subsection AsyncNDRegisterAccessor_details Design decisions and implementation details
+
+- The data transport queue is of type UserType, so the potentially expensive and slow operations (data readout,
+  de-multiplexing, data conversion) are happening only once, before the data is distributed to multiple subscribers.
+- The additional send buffer is necessary because data has to be send through the queue in this container, and usually
+  the data is not provided in this format.
+  Even if the NDRegisterAccessor would provide the user buffer in this format it could not be used here, because it is
+  used by the receiving side which is running in another thread.
+- To unsubscribe, the NDRegisterAccessor is calling AsyncAccessorManager::unsibscribe() with its TransferElementID from
+  the destructor.
+  For the unsubscribe mechanism to work, the AsyncAccessorManager can only hold a weak pointer.
+  If it would hold a shared pointer, the destructor would never be called because this instance survives,
+  even if the user code has deleted all copies of the AsyncNDRegisterAccessor.
+- To manage the weak pointer and to have an entry object for the implementation, the helper class
```

Comment:
The justification for the AsyncVariable is too low level for my taste. It is better to explain what it does rather then why you felt it was necessary to introduce it. I still don't really understand what it does. Does it represent a single AsyncNDRegisterAccessor inside the AsyncAccessorManager? I think, this is all what matters here.

Reply:
Yes, in the new design it reprecents one AsyncNDRegisterAccessor